modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteRepositoryFactory.java

```diff
@@ -193,6 +207,26 @@ public IgniteRepositoryFactory(String springCfgPath, ApplicationContext ctx) {
         });
     }
 
+    /** {@inheritDoc} */
+    @Override protected RepositoryComposition.RepositoryFragments getRepositoryFragments(RepositoryMetadata metadata) {
+        RepositoryComposition.RepositoryFragments fragments = RepositoryComposition.RepositoryFragments.empty();
+
+        boolean isQryDslRepo = QUERY_DSL_PRESENT
```

Comment:
This code will likely not work if dependencies are optional, needs to be tested additionally.

Reply:
Why it will not work, `QUERY_DSL_PRESENT` contains result of:

```java
org.springframework.util.ClassUtils
			.isPresent("com.querydsl.core.types.Predicate", QuerydslUtils.class.getClassLoader());
```

The spring-data-jpa also use this way to detect the query dsl: https://github.com/spring-projects/spring-data-jpa/blob/master/src/main/java/org/springframework/data/jpa/repository/support/JpaRepositoryFactory.java#L241